sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/CoreMessageSender.java

```diff
@@ -269,6 +277,29 @@ private void scheduleSendTimeout(SendWorkItem<DeliveryState> sendWorkItem) {
         // Timer to timeout the request
         ScheduledFuture<?> timeoutTask = Timer.schedule(() -> {
             if (!sendWorkItem.getWork().isDone()) {
+            	if (sendWorkItem.getTimeoutTracker().originalTimeout().compareTo(MINIMUM_SEND_TIMEOUT_TO_CONSIDER_AS_NETWORK_OUTAGE) >= 0) {
+            		if (this.isInConsecutiveTimeoutState) {
+                		Duration consecutiveTimeoutsDuration = Duration.between(this.consecutiveTimeoutsStartedAt, Instant.now());
+                		if (consecutiveTimeoutsDuration.compareTo(MAXIMUM_CONSECUTIVE_TIMEOUT_DURATION) > 0) {
+                			// Consecutive timeouts. Reconnect by forcing a connection error
+                			try {
+                                this.underlyingFactory.scheduleOnReactorThread(new DispatchHandler() {
+                                    @Override
+                                    public void onEvent() {
+                                        CoreMessageSender.this.underlyingFactory.onConnectionError(new ErrorCondition());
```

Comment:
>onConnectionError(new ErrorCondition() [](start = 81, length = 38)

Will this show up in logs as empty error? If yes, could you add some error message which makes it obvious?